src/common/src/cache.rs

```diff
@@ -679,21 +685,31 @@ impl<K: LruKey, T: LruValue> LruCache<K, T> {
     ) -> CacheableEntry<K, T> {
         let mut to_delete = vec![];
         // Drop the entries outside lock to avoid deadlock.
+        let mut senders = vec![];
         let handle = unsafe {
             let mut shard = self.shards[self.shard(hash)].lock();
             let pending_request = shard.write_request.remove(&key);
             let ptr = shard.insert(key, hash, charge, value, &mut to_delete);
             debug_assert!(!ptr.is_null());
-            if let Some(que) = pending_request {
-                for sender in que {
-                    let _ = sender.send(());
-                }
+            if let Some(mut que) = pending_request {
+                (*ptr).add_many_refs(que.len() as u32);
+                senders = std::mem::take(&mut que);
             }
             CacheableEntry {
                 cache: self.clone(),
                 handle: ptr,
             }
         };
+        let mut errs = vec![];
```

Comment:
> We don't have to collect the `errs` here because we are sending the entries outside the lock.

in the future we may handle the destruction of errs together to accelerate

Reply:
What is the purpose of collecting errors?

Reply:
> What is the purpose of collecting errors?

ok, I have just removed this part of code